hpat/datatypes/hpat_pandas_series_functions.py

```
@@ -2218,3 +2218,136 @@ def hpat_pandas_series_median_impl(self, axis=None, skipna=True, level=None, num
+        values to forward/backward fill.
+        *unsupported*
+    downcast: :obj:`dict` or :obj:`string` {`infer`}, default None
+        Controls logic of downcasting elements to particular dtype
+        *unsupported*
+    Returns
+    -------
+    :obj:`pandas.Series`
+         returns :obj:`pandas.Series` with missed values filled.
+    """
+
+    _func_name = 'Method fillna().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+
+    if not (isinstance(axis, (types.Integer, types.UnicodeType, types.Omitted)) or axis is None):
+        raise TypingError('{} The axis must be an Integer or String. Given: {}'.format(_func_name, axis))
+
+    if not (isinstance(inplace, types.Literal) and isinstance(inplace, types.Boolean)
+            or isinstance(inplace, types.Omitted)
+            or inplace == False):
+        raise TypingError('{} The inplace must be a literal Boolean constant. Given: {}'.format(_func_name, inplace))
+
+    if not ((method is None or isinstance(method, types.Omitted))
+            and (limit is None or isinstance(limit, types.Omitted))
+            and (downcast is None or isinstance(downcast, types.Omitted))
+    ):
+        raise TypingError('{} Unsupported parameters. Given method: {}, limit: {}, downcast: {}'.format(
+                _func_name, method, limit, downcast))
+
+    # inplace value has to be known at compile time to select between implementations with different signatures
+    if ((isinstance(inplace, types.Literal) and inplace.literal_value == True)
+        or (isinstance(inplace, bool) and inplace == True)):
+        # do operation inplace, fill the NA/NaNs in the same array and return None
+        if isinstance(self.dtype, types.UnicodeType):
+            # TODO: StringArrayType cannot resize inplace, and assigning a copy back to self._data is not possible now
+            return None
```

Maybe better to raise exception here something like `unsupported...`?

---

Added